src/io/output.rs

```diff
@@ -2,83 +2,159 @@ use cairo_vm::types::relocatable::MaybeRelocatable;
 
 #[derive(Debug)]
 pub struct StarknetOsOutput {
     /// The state commitment before this block.
-    pub prev_state_root: Felt252,
+    pub initial_root: Felt252,
     /// The state commitment after this block.
-    pub new_state_root: Felt252,
+    pub final_root: Felt252,
     /// The number (height) of this block.
     pub block_number: Felt252,
     /// The hash of this block.
     pub block_hash: Felt252,
     /// The Starknet chain config hash
-    pub config_hash: Felt252,
+    pub starknet_os_config_hash: Felt252,
+    /// Whether KZG data availability was used.
+    pub use_kzg_da: Felt252,
     /// List of messages sent to L1 in this block
     pub messages_to_l1: Vec<Felt252>,
     /// List of messages from L1 handled in this block
     pub messages_to_l2: Vec<Felt252>,
     /// List of the storage updates.
-    pub state_updates: Vec<Felt252>,
+    pub contracts: Vec<Felt252>,
     /// List of the newly declared contract classes.
-    pub contract_class_diff: Vec<Felt252>,
+    pub classes: Vec<Felt252>,
 }
 
 impl StarknetOsOutput {
     pub fn from_run(vm: &VirtualMachine) -> Result<Self, SnOsError> {
-        let builtin_end_ptrs = vm.get_return_values(8).map_err(|e| SnOsError::CatchAll(e.to_string()))?;
-        let output_base = vm
-            .get_builtin_runners()
-            .iter()
-            .find(|&elt| matches!(elt, BuiltinRunner::Output(_)))
-            .expect("Os vm should have the output builtin")
-            .base();
-        let size_bound_up = match builtin_end_ptrs.last().unwrap() {
-            MaybeRelocatable::Int(val) => val,
-            _ => return Err(SnOsError::CatchAll("value should be an int".to_string())),
-        };
-
-        // Get is input and check that everything is an integer.
-        let size = felt_vm2usize(Some(&(*size_bound_up - Felt252::from(output_base))))?;
-        let raw_output = vm.get_range((output_base as isize, 0).into(), size);
-        let raw_output: Vec<Felt252> = raw_output
-            .iter()
-            .map(|x| {
-                if let MaybeRelocatable::Int(val) = x.clone().unwrap().into_owned() {
-                    val
-                } else {
-                    panic!("Output should be all integers")
-                }
-            })
-            .collect();
-
-        decode_output(raw_output)
+        let (output_base, output_size) = get_output_info(vm)?;
+        let raw_output = get_raw_output(vm, output_base, output_size)?;
+        decode_output(raw_output.into_iter())
     }
 }
 
-pub fn decode_output(mut os_output: Vec<Felt252>) -> Result<StarknetOsOutput, SnOsError> {
-    let header: Vec<Felt252> = os_output.drain(..HEADER_SIZE).collect();
+/// Gets the output base segment and the output size from the VM return values and the VM
+/// output builtin.
+fn get_output_info(vm: &VirtualMachine) -> Result<(usize, usize), SnOsError> {
+    let builtin_end_ptrs = vm.get_return_values(8).map_err(|e| SnOsError::CatchAll(e.to_string()))?;
```

Comment:
`8` should be a constant, I think it probably comes from the number of builtins, which I think you can query for